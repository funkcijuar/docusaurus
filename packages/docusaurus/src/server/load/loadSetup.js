/**
 * Copyright (c) 2017-present, Facebook, Inc.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import path from 'path';
import load from '../index';

// Helper methods to setup dummy/fake projects
const loadSetup = async name => {
  const fixtures = path.join(__dirname, '__tests__', '__fixtures__');
  const simpleSite = path.join(fixtures, 'simple-site');
  const customSite = path.join(fixtures, 'custom-site');

  switch (name) {
    case 'simple':
      return load(simpleSite);
    case 'custom':
      return load(customSite);
    default:
      return {};
  }
};

export default loadSetup;
