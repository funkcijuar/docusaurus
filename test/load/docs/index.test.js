import loadDocs from '@lib/load/docs';
import loadSetup from '../../loadSetup';

describe('loadDocs', () => {
  test('simple website', async () => {
    const props = await loadSetup('simple');
    const {siteDir, docsDir, env} = props;
    const docsData = await loadDocs({siteDir, docsDir, env});
    expect(docsData).toMatchSnapshot();
  });

  test('versioned website', async () => {
    const props = await loadSetup('versioned');
    const {siteDir, docsDir, env} = props;
    const docsData = await loadDocs({siteDir, docsDir, env});
    expect(docsData).toMatchSnapshot();
  });

  test('versioned & translated website', async () => {
    const props = await loadSetup('transversioned');
    const {siteDir, docsDir, env} = props;
    const docsData = await loadDocs({siteDir, docsDir, env});
    expect(docsData).toMatchSnapshot();
  });

  test('translated website', async () => {
    const props = await loadSetup('translated');
    const {siteDir, docsDir, env} = props;
    const docsData = await loadDocs({siteDir, docsDir, env});
    expect(docsData).toMatchSnapshot();
  });
});
