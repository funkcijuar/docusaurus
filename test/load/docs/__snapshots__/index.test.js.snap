// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loadDocs simple website 1`] = `
Object {
  "foo/bar": Object {
    "id": "foo/bar",
    "language": undefined,
    "localized_id": "foo/bar",
    "permalink": "docs/foo/bar",
    "source": "foo/bar.md",
    "title": "Bar",
    "version": undefined,
  },
  "foo/baz": Object {
    "id": "foo/baz",
    "language": undefined,
    "localized_id": "foo/baz",
    "permalink": "docs/foo/baz",
    "source": "foo/baz.md",
    "title": "baz",
    "version": undefined,
  },
  "hello": Object {
    "id": "hello",
    "language": undefined,
    "localized_id": "hello",
    "permalink": "docs/hello",
    "source": "hello.md",
    "title": "Hello, World !",
    "version": undefined,
  },
}
`;

exports[`loadDocs translated website 1`] = `
Object {
  "en-hello": Object {
    "id": "en-hello",
    "language": "en",
    "localized_id": "en-hello",
    "permalink": "docs/en/hello",
    "source": "hello.md",
    "title": "Hello, World !",
    "version": undefined,
  },
  "foo/bar": Object {
    "id": "foo/bar",
    "language": undefined,
    "localized_id": "foo/bar",
    "permalink": "docs/foo/bar",
    "source": "foo/bar.md",
    "title": "Bar",
    "version": undefined,
  },
  "foo/baz": Object {
    "id": "foo/baz",
    "language": undefined,
    "localized_id": "foo/baz",
    "permalink": "docs/foo/baz",
    "source": "foo/baz.md",
    "title": "baz",
    "version": undefined,
  },
  "ko-docs/foo/bar": Object {
    "id": "ko-docs/foo/bar",
    "language": "ko",
    "localized_id": "ko-docs/foo/bar",
    "permalink": "docs/ko/docs/foo/bar",
    "source": "ko/docs/foo/bar.md",
    "title": "Bar",
    "version": undefined,
  },
  "ko-docs/foo/baz": Object {
    "id": "ko-docs/foo/baz",
    "language": "ko",
    "localized_id": "ko-docs/foo/baz",
    "permalink": "docs/ko/docs/foo/baz",
    "source": "ko/docs/foo/baz.md",
    "title": "baz",
    "version": undefined,
  },
  "ko-docs/hello": Object {
    "id": "ko-docs/hello",
    "language": "ko",
    "localized_id": "ko-docs/hello",
    "permalink": "docs/ko/docs/hello",
    "source": "ko/docs/hello.md",
    "title": "Hello, World !",
    "version": undefined,
  },
}
`;

exports[`loadDocs versioned & translated website 1`] = `
Object {
  "en-hello": Object {
    "id": "en-hello",
    "language": "en",
    "localized_id": "en-hello",
    "permalink": "docs/en/next/hello",
    "source": "hello.md",
    "title": "Hello, World !",
    "version": "next",
  },
  "foo/bar": Object {
    "category": "Test",
    "id": "foo/bar",
    "language": undefined,
    "localized_id": "foo/bar",
    "next": "foo/baz",
    "next_id": "foo/baz",
    "next_title": "baz",
    "permalink": "docs/next/foo/bar",
    "sidebar": "docs",
    "source": "foo/bar.md",
    "title": "Bar",
    "version": "next",
  },
  "foo/baz": Object {
    "category": "Test",
    "id": "foo/baz",
    "language": undefined,
    "localized_id": "foo/baz",
    "next": "hello",
    "next_id": "hello",
    "next_title": "Next",
    "permalink": "docs/next/foo/baz",
    "previous": "foo/bar",
    "previous_id": "foo/bar",
    "previous_title": "Bar",
    "sidebar": "docs",
    "source": "foo/baz.md",
    "title": "baz",
    "version": "next",
  },
  "ko-version-1.0.0-version-1.0.0/foo/bar": Object {
    "id": "ko-version-1.0.0-version-1.0.0/foo/bar",
    "language": "ko",
    "localized_id": "ko-version-1.0.0-version-1.0.0/foo/bar",
    "permalink": "docs/ko/1.0.0/version-1.0.0/foo/bar",
    "source": "ko/version-1.0.0/foo/bar.md",
    "title": "Bar",
    "version": "1.0.0",
  },
  "ko-version-1.0.0-version-1.0.0/foo/baz": Object {
    "id": "ko-version-1.0.0-version-1.0.0/foo/baz",
    "language": "ko",
    "localized_id": "ko-version-1.0.0-version-1.0.0/foo/baz",
    "permalink": "docs/ko/1.0.0/version-1.0.0/foo/baz",
    "source": "ko/version-1.0.0/foo/baz.md",
    "title": "baz",
    "version": "1.0.0",
  },
  "ko-version-1.0.0-version-1.0.0/hello": Object {
    "id": "ko-version-1.0.0-version-1.0.0/hello",
    "language": "ko",
    "localized_id": "ko-version-1.0.0-version-1.0.0/hello",
    "permalink": "docs/ko/1.0.0/version-1.0.0/hello",
    "source": "ko/version-1.0.0/hello.md",
    "title": "Hello, World !",
    "version": "1.0.0",
  },
  "ko-version-1.0.1-version-1.0.1/foo/bar": Object {
    "id": "ko-version-1.0.1-version-1.0.1/foo/bar",
    "language": "ko",
    "localized_id": "ko-version-1.0.1-version-1.0.1/foo/bar",
    "permalink": "docs/ko/version-1.0.1/foo/bar",
    "source": "ko/version-1.0.1/foo/bar.md",
    "title": "Bar",
    "version": "1.0.1",
  },
  "ko-version-1.0.1-version-1.0.1/foo/baz": Object {
    "id": "ko-version-1.0.1-version-1.0.1/foo/baz",
    "language": "ko",
    "localized_id": "ko-version-1.0.1-version-1.0.1/foo/baz",
    "permalink": "docs/ko/version-1.0.1/foo/baz",
    "source": "ko/version-1.0.1/foo/baz.md",
    "title": "baz",
    "version": "1.0.1",
  },
  "ko-version-1.0.1-version-1.0.1/hello": Object {
    "id": "ko-version-1.0.1-version-1.0.1/hello",
    "language": "ko",
    "localized_id": "ko-version-1.0.1-version-1.0.1/hello",
    "permalink": "docs/ko/version-1.0.1/hello",
    "source": "ko/version-1.0.1/hello.md",
    "title": "Hello, World !",
    "version": "1.0.1",
  },
  "version-1.0.0-foo/bar": Object {
    "category": "Test",
    "id": "version-1.0.0-foo/bar",
    "language": undefined,
    "localized_id": "version-1.0.0-foo/bar",
    "next": "version-1.0.0-foo/baz",
    "next_id": "version-1.0.0-foo/baz",
    "next_title": "Baz",
    "permalink": "docs/1.0.0/foo/bar",
    "sidebar": "version-1.0.0-docs",
    "source": "version-1.0.0/foo/bar.md",
    "title": "Bar",
    "version": "1.0.0",
  },
  "version-1.0.0-foo/baz": Object {
    "category": "Test",
    "id": "version-1.0.0-foo/baz",
    "language": undefined,
    "localized_id": "version-1.0.0-foo/baz",
    "next": "version-1.0.0-hello",
    "next_id": "version-1.0.0-hello",
    "next_title": "Hello, World !",
    "permalink": "docs/1.0.0/foo/baz",
    "previous": "version-1.0.0-foo/bar",
    "previous_id": "version-1.0.0-foo/bar",
    "previous_title": "Bar",
    "sidebar": "version-1.0.0-docs",
    "source": "version-1.0.0/foo/baz.md",
    "title": "Baz",
    "version": "1.0.0",
  },
  "version-1.0.0-hello": Object {
    "category": "Guides",
    "id": "version-1.0.0-hello",
    "language": undefined,
    "localized_id": "version-1.0.0-hello",
    "permalink": "docs/1.0.0/hello",
    "previous": "version-1.0.0-foo/baz",
    "previous_id": "version-1.0.0-foo/baz",
    "previous_title": "Baz",
    "sidebar": "version-1.0.0-docs",
    "source": "version-1.0.0/hello.md",
    "title": "Hello, World !",
    "version": "1.0.0",
  },
  "version-1.0.1-foo/bar": Object {
    "category": "Test",
    "id": "version-1.0.1-foo/bar",
    "language": undefined,
    "localized_id": "version-1.0.1-foo/bar",
    "next": "version-1.0.1-foo/baz",
    "next_id": "version-1.0.1-foo/baz",
    "next_title": "Baz",
    "permalink": "docs/foo/bar",
    "sidebar": "version-1.0.1-docs",
    "source": "version-1.0.1/foo/bar.md",
    "title": "Bar",
    "version": "1.0.1",
  },
  "version-1.0.1-foo/baz": Object {
    "category": "Test",
    "id": "version-1.0.1-foo/baz",
    "language": undefined,
    "localized_id": "version-1.0.1-foo/baz",
    "next": "version-1.0.1-hello",
    "next_id": "version-1.0.1-hello",
    "next_title": "Hello, World !",
    "permalink": "docs/foo/baz",
    "previous": "version-1.0.1-foo/bar",
    "previous_id": "version-1.0.1-foo/bar",
    "previous_title": "Bar",
    "sidebar": "version-1.0.1-docs",
    "source": "version-1.0.1/foo/baz.md",
    "title": "Baz",
    "version": "1.0.1",
  },
  "version-1.0.1-hello": Object {
    "category": "Guides",
    "id": "version-1.0.1-hello",
    "language": undefined,
    "localized_id": "version-1.0.1-hello",
    "permalink": "docs/hello",
    "previous": "version-1.0.1-foo/baz",
    "previous_id": "version-1.0.1-foo/baz",
    "previous_title": "Baz",
    "sidebar": "version-1.0.1-docs",
    "source": "version-1.0.1/hello.md",
    "title": "Hello, World !",
    "version": "1.0.1",
  },
}
`;

exports[`loadDocs versioned website 1`] = `
Object {
  "foo/bar": Object {
    "id": "foo/bar",
    "language": undefined,
    "localized_id": "foo/bar",
    "permalink": "docs/next/foo/bar",
    "source": "foo/bar.md",
    "title": "Bar",
    "version": "next",
  },
  "foo/baz": Object {
    "id": "foo/baz",
    "language": undefined,
    "localized_id": "foo/baz",
    "permalink": "docs/next/foo/baz",
    "source": "foo/baz.md",
    "title": "baz",
    "version": "next",
  },
  "hello": Object {
    "id": "hello",
    "language": undefined,
    "localized_id": "hello",
    "permalink": "docs/next/hello",
    "source": "hello.md",
    "title": "Hello, World !",
    "version": "next",
  },
  "version-1.0.0-foo/bar": Object {
    "category": "Test",
    "id": "version-1.0.0-foo/bar",
    "language": undefined,
    "localized_id": "version-1.0.0-foo/bar",
    "next": "version-1.0.0-foo/baz",
    "next_id": "version-1.0.0-foo/baz",
    "next_title": "Baz",
    "permalink": "docs/1.0.0/foo/bar",
    "sidebar": "version-1.0.0-docs",
    "source": "version-1.0.0/foo/bar.md",
    "title": "Bar",
    "version": "1.0.0",
  },
  "version-1.0.0-foo/baz": Object {
    "category": "Test",
    "id": "version-1.0.0-foo/baz",
    "language": undefined,
    "localized_id": "version-1.0.0-foo/baz",
    "next": "version-1.0.0-hello",
    "next_id": "version-1.0.0-hello",
    "next_title": "Hello, World !",
    "permalink": "docs/1.0.0/foo/baz",
    "previous": "version-1.0.0-foo/bar",
    "previous_id": "version-1.0.0-foo/bar",
    "previous_title": "Bar",
    "sidebar": "version-1.0.0-docs",
    "source": "version-1.0.0/foo/baz.md",
    "title": "Baz",
    "version": "1.0.0",
  },
  "version-1.0.0-hello": Object {
    "category": "Guides",
    "id": "version-1.0.0-hello",
    "language": undefined,
    "localized_id": "version-1.0.0-hello",
    "permalink": "docs/1.0.0/hello",
    "previous": "version-1.0.0-foo/baz",
    "previous_id": "version-1.0.0-foo/baz",
    "previous_title": "Baz",
    "sidebar": "version-1.0.0-docs",
    "source": "version-1.0.0/hello.md",
    "title": "Hello, World !",
    "version": "1.0.0",
  },
  "version-1.0.1-foo/bar": Object {
    "category": "Test",
    "id": "version-1.0.1-foo/bar",
    "language": undefined,
    "localized_id": "version-1.0.1-foo/bar",
    "next": "version-1.0.1-foo/baz",
    "next_id": "version-1.0.1-foo/baz",
    "next_title": "Baz",
    "permalink": "docs/foo/bar",
    "sidebar": "version-1.0.1-docs",
    "source": "version-1.0.1/foo/bar.md",
    "title": "Bar",
    "version": "1.0.1",
  },
  "version-1.0.1-foo/baz": Object {
    "category": "Test",
    "id": "version-1.0.1-foo/baz",
    "language": undefined,
    "localized_id": "version-1.0.1-foo/baz",
    "next": "version-1.0.1-hello",
    "next_id": "version-1.0.1-hello",
    "next_title": "Hello, World !",
    "permalink": "docs/foo/baz",
    "previous": "version-1.0.1-foo/bar",
    "previous_id": "version-1.0.1-foo/bar",
    "previous_title": "Bar",
    "sidebar": "version-1.0.1-docs",
    "source": "version-1.0.1/foo/baz.md",
    "title": "Baz",
    "version": "1.0.1",
  },
  "version-1.0.1-hello": Object {
    "category": "Guides",
    "id": "version-1.0.1-hello",
    "language": undefined,
    "localized_id": "version-1.0.1-hello",
    "permalink": "docs/hello",
    "previous": "version-1.0.1-foo/baz",
    "previous_id": "version-1.0.1-foo/baz",
    "previous_title": "Baz",
    "sidebar": "version-1.0.1-docs",
    "source": "version-1.0.1/hello.md",
    "title": "Hello, World !",
    "version": "1.0.1",
  },
}
`;
